opensearch-project/OpenSearch-Dashboards
# [BUG] Fix bug in data source aggregated view to change it to depend on displayAllCompatibleDataSources property to show the badge value

### Description
This change fixes bug in data source aggregated view to change it to depend on displayAllCompatibleDataSources property to show the badge value

### Issues Resolved


## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/17714150/1b1d17b3-ecf6-471a-99f9-e0a9b0c045a0



## Testing the changes
The following was performed in the recording above:
1. enable data source
2. go to maps plugin list page, and should see the data source component mounted correctly and display all data sources
3. go to maps detail page, and should see the badge number which is the number of data source being used in all layers, add a layer, should see the badge number is updated correctly, remove a layer should reduce 1 in the badge number

### Check List
